### PR TITLE
Detect invalid .well-known endpoint response

### PR DESCRIPTION
**Pull request by alyaeanyx**
As discussed on Matrix.

### CODE REVIEW

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
@poljar any objections?

File: crates/matrix-sdk/src/client/builder.rs [hunk 1]
```diff
@@ -312,7 +312,11 @@ impl ClientBuilder {
                         None,
                         [MatrixVersion::V1_0].into_iter().collect(),
                     )
-                    .await?;
+                    .await
+                    .map_err(|e| match e {
+                        HttpError::ClientApi(_) => ClientBuildError::AutoDiscovery,
```
Comment (poljar) [inline crates/matrix-sdk/src/client/builder.rs:317]: This seems sensible, but don't we want to just move the inner error from `ClientApi` into `AutoDiscovery`? Why throw it away.

Comment (jplatte) [reply crates/matrix-sdk/src/client/builder.rs:317]: It's most likely not going to contain anything interesting, but I wouldn't object to moving it 🤷🏼

Comment (poljar) [reply crates/matrix-sdk/src/client/builder.rs:317]: I agree, but it's more likely that someone will complain that the error has been thrown away than complain that the error has been preserved.

Comment (alyaeanyx) [reply crates/matrix-sdk/src/client/builder.rs]: You're right, the underlying error is preserved now.

**Review by poljar — APPROVED** [review]

**Comment by poljar** [comment]
Can we just fix the style so CI is happy, after that we can merge.

**Comment by alyaeanyx** [comment]
Should work now. Probably best to squash the commits into one.

**Comment by jplatte** [comment]
Seems like GitHub is unable to fix itself here. I'll close and reopen to trigger CI.

**Pull request merged**